pilot/pkg/model/push_context.go

```diff
@@ -763,6 +779,53 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthNPolicies(env *Environment) error {
+	authNPolicies, err := env.List(schemas.AuthenticationPolicy.Type, NamespaceAll)
+	if err != nil {
+		return err
+	}
+
+	sortConfigByCreationTime(authNPolicies)
+	ps.AuthNPolicies = map[host.Name]map[int][]*Config{}
+
+	for i, spec := range authNPolicies {
+		policy := spec.Spec.(*authn.Policy)
+		if len(policy.Targets) > 0 {
+			for _, dest := range policy.Targets {
+				hostName := ResolveShortnameToFQDN(dest.Name, spec.ConfigMeta)
+				if ps.AuthNPolicies[hostName] == nil {
+					ps.AuthNPolicies[hostName] = map[int][]*Config{}
+				}
+				if len(dest.Ports) > 0 {
+					for _, port := range dest.Ports {
+						ps.AuthNPolicies[hostName][int(port.GetNumber())] = append(ps.AuthNPolicies[hostName][int(port.GetNumber())], &authNPolicies[i])
+					}
+				} else {
+					ps.AuthNPolicies[hostName][noPortProvided] = append(ps.AuthNPolicies[hostName][noPortProvided], &authNPolicies[i])
+				}
+
+			}
+		} else {
+			// if no targets provided, store at namespace level
+			if ps.AuthNPolicies[host.Name(spec.Namespace)] == nil {
+				ps.AuthNPolicies[host.Name(spec.Namespace)] = map[int][]*Config{}
+			}
+			ps.AuthNPolicies[host.Name(spec.Namespace)][noPortProvided] = append(ps.AuthNPolicies[host.Name(spec.Namespace)][noPortProvided], &authNPolicies[i])
+		}
+	}
+
+	if specs, err := env.List(schemas.AuthenticationMeshPolicy.Type, ""); err == nil {
```

Comment:
no need to list, use get 
`env.Get(schemas.AuthenticationMeshPolicy.Type,  constants.DefaultAuthenticationPolicyName , "")`

Reply:
this suggested fix for reason breaks the unit tests I have in place.  It also breaks the controlplane in e2e tests - pilot logs become riddled with:
```
2019-09-17T13:59:44.003211Z	info	mcp	(re)trying to establish new MCP sink stream
2019-09-17T13:59:44.003335Z	error	mcp	Failed to create a new MCP sink stream: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial tcp: lookup istio-galley.istio-system.svc on 172.21.0.10:53: no such host"
2019-09-17T13:59:44.025049Z	info	Configuration not synced: first push for [istio/networking/v1alpha3/serviceentries istio/networking/v1alpha3/destinationrules istio/mixer/v1/config/client/quotaspecbindings istio/rbac/v1alpha1/servicerolebindings istio/rbac/v1alpha1/clusterrbacconfigs istio/networking/v1alpha3/envoyfilters istio/networking/v1alpha3/sidecars istio/authentication/v1alpha1/policies istio/rbac/v1alpha1/serviceroles istio/security/v1beta1/authorizationpolicies istio/networking/v1alpha3/gateways istio/config/v1alpha2/httpapispecs istio/config/v1alpha2/httpapispecbindings istio/authentication/v1alpha1/meshpolicies istio/networking/v1alpha3/virtualservices istio/mixer/v1/config/client/quotaspecs istio/rbac/v1alpha1/rbacconfigs] not received
```
It appears the default mesh policy is never found for some reason with `env.Get()`.  Seeing as how the current implementation for getting the default mesh policy is the same as in [configStore](https://github.com/istio/istio/blob/master/pilot/pkg/model/config.go#L618) - I don't think it needs to be addressed in this PR.  @hzxuzhonghu you can open a separate issue for this if you feel it is a bug?